client/me/concierge/controller.js

```diff
@@ -28,6 +30,18 @@ const book = ( context, next ) => {
 	next();
 };
 
+const cancel = ( context, next ) => {
+	context.primary = (
+		<ConciergeMain
```

Comment:
Is there any reason the cancellation page needs to go through `ConciergeMain`? Since it's a single step, you don't gain any benefit from the multi-step aspects. And [as mentioned on another PR](https://github.com/Automattic/wp-calypso/pull/21364#pullrequestreview-87974801), `ConciergeMain` blocks you showing the Skeleton view until sites and available times data is fetched — which this page doesn't even need, so that just slows the cancellation action down.

I think you could instead render the `CancelConfirmationStep` on its own, and that component could show a placeholder while the cancellation action fires, showing success or error based on the result. See my last comment on 494-gh-hg